File: LeftRightTrapMid/LeftRightTrapMid.cs
using System;

namespace CalculusIntegrationTechniques
{
    class SimpsonsRule
    {
        public const double M_PI = 3.14159265358979323846;

        static void Main(string[] args)
        {
            double a = 0;
            double b = M_PI;
            int iterations = 1000;

            double dx = (b - a) / iterations;

            double hL = 0;
            double hR = 0;
            double hM = 0;
            double hT = 0;

            double xL = 0;
            double xR = 0;
            double xT = 0;
            double xM = 0;

            int i = 0;
            for (i = 1; i < iterations; i++)
            {
                // Left Rectangular Rule
                xL = a + (i - 1) * dx;
                hL = hL + f(xL);

                // Right Rectangular Rule
                xR = a + (i) * dx;
                hR = hR + f(xR);

                // Midpoint Rule
                xM = a + (xL + xR) / 2;
                hM = hM + f(xM);

                // Trapezoidal Rule
                xT = (f(xL) + f(xR)) / 2;
                hT = hT + xT;
            }

            double LeftRectangularResult = hL * dx;
            double RightRectangularResult = hR * dx;
            double MidPointResult = hM * dx;
            double TrapezoidalResult = hT * dx;

            Console.WriteLine($"Left Rectangular Result: {LeftRectangularResult}");
            Console.WriteLine($"Right Rectangular Result: {RightRectangularResult}");
            Console.WriteLine($"MidPoint Result: {MidPointResult}");
            Console.WriteLine($"Trapezoidal Result: {TrapezoidalResult}");
        }

        private static double f(double x)
        {
            return x * x;
        }
    }
}

File: LeftRightTrapMid/LeftRightTrapMid.py
a = 0
b = 3.14159265358979323846
iterations = 1000

dx = (b - a) / iterations

hL = 0
hR = 0
hM = 0
hT = 0


def f(x):
    return x * x


i = 1
while i < iterations:
    xL = a + (i - 1) * dx
    hL += + f(xL)
    xR = a + i * dx
    hR += + f(xR)
    xM = a + (xL + xR) / 2
    hM += + f(xM)
    xT = (f(xL) + f(xR)) / 2
    hT += + xT
    i += 1

LeftRectangularResult = hL * dx
RightRectangularResult = hR * dx
MidPointResult = hM * dx
TrapezoidalResult = hT * dx

print("Left Rectangular Result: %.5f\n" % LeftRectangularResult)
print("Right Rectangular Result: %.5f\n" % RightRectangularResult)
print("MidPoint Result: %.5f\n" % MidPointResult)
print("Trapezoidal Result: %.5f\n" % TrapezoidalResult)
